cpu/src/lib.rs

```diff
@@ -391,23 +356,24 @@ where
 {
     const OPCODE: u32 = READ_ADVICE;
 
-    fn execute(state: &mut M, ops: Operands<i32>) {
-        // Advice tape location
-        let addr = ops.a() as u32;
-        let buf_len = ops.b() as u32;
+    fn execute(_state: &mut M, _ops: Operands<i32>) {
+        panic!("execute_with_advice should be called instead");
+    }
 
-        // Memory location
-        let mem_addr = ops.c();
+    fn execute_with_advice<Adv>(state: &mut M, ops: Operands<i32>, advice: &mut Adv)
+    where
+        M: MachineWithCpuChip,
+        Adv: AdviceProvider,
+    {
+        let clk = state.cpu().clock;
+        let fp = state.cpu().fp as i32;
+        let mem_addr = fp + ops.c();
```

Comment:
I had a question about this - after changing the Fibonacci assembly to read

    advread -8

the offset `-8` seems to end up in operand `c`, hence `ops.c()` here. Is that intended, or should it be in operand `a`?

Reply:
Ah, it's because the assembler is not run automatically when running the tests. We may want to add that. The offset should show up in the first operand in the new binary.